subprojects/launcher/src/main/java/org/gradle/tooling/internal/provider/TestExecutionRequestAction.java

```diff
@@ -53,7 +58,17 @@ public static TestExecutionRequestAction create(BuildClientSubscriptions clientS
         return new TestExecutionRequestAction(clientSubscriptions, startParameter,
                                                 ImmutableSet.copyOf(testExecutionRequest.getTestExecutionDescriptors()),
                                                 ImmutableSet.copyOf(testClassNames),
-                                                providerInternalJvmTestRequests);
+                                                providerInternalJvmTestRequests,
+                                                getDebugOptions(testExecutionRequest));
+    }
+
+    private static InternalDebugOptions getDebugOptions(ProviderInternalTestExecutionRequest testExecutionRequest) {
+        try {
+            return testExecutionRequest.getDebugOptions();
+        } catch (UnsupportedMethodException e) {
+            // default value for older Gradle clients
+            return new DefaultDebugOptions();
```

Comment:
⭕️ Nit: This seems a bit opaque that we create the `DefaultDebugOptions` and it "magically" has default values. I feel like the default values should be explicit here.

Reply:
Ah yes, this POJO type that transfers the configuration from the TAPI client to the daemon. Basically, this just says that if an old TAPI client is connected then don't enable debugging.